### PR TITLE
chore: prune expired consensus states on duplicate header updates

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Moving call to consensus state pruning function above duplicate header check
- Adding test case

closes: #1207 

### Commit Message / Changelog Entry

```bash
chore: prune expired `07-tendermint` consensus states on duplicate header updates
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/07-tendermint/update_test.go [hunk 1]
```diff
@@ -420,6 +420,43 @@ func (suite *TendermintTestSuite) TestUpdateState() {
 				suite.Require().False(found)
 			}, true,
 		},
+		{
+			"success with pruned consensus state using duplicate header", func() {
+				// this height will be expired and pruned
+				err := path.EndpointA.UpdateClient()
+				suite.Require().NoError(err)
+				pruneHeight = path.EndpointA.GetClientState().GetLatestHeight().(clienttypes.Height)
```
Comment (charleenfei) [inline modules/light-clients/07-tendermint/update_test.go:428]: maybe first grab the consensus state here before it's pruned?

**Review by charleenfei — APPROVED** [review]
one nit but otherwise lgtm!

File: modules/light-clients/07-tendermint/update_test.go (hunk 1 above)
Comment (damiannolan) [reply modules/light-clients/07-tendermint/update_test.go:428]: do you mean just to assert that it exists at the pruneHeight? 
I can push an update with that!

Comment (damiannolan) [reply modules/light-clients/07-tendermint/update_test.go:428]: done ✅

**Pull request merged**